File: ApplicationStock/ApplicationConsole/ProjectStockLibrary/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectStockLibrary
{
    public  class Order
    {
        [Key]
        public Guid _id { get; private set; }
        public string _orderName { get;  set; }
        public  DateTime _orderDate { get;  set; }
        [ForeignKey("Stock")]
        private Stock _stock { get; set; }
        public  int _nbStock { get;  set; }
        public Order(string orderName,Stock stock,int nbStock) 
        {
            _orderName = orderName;
            _orderDate = DateTime.Now;
            _id = Guid.NewGuid();
            _stock = stock;
            _nbStock = nbStock;

        }

        public Order(string orderName, int nbStock)
        {
            _orderName = orderName;
            _orderDate = DateTime.Now;
            _id = Guid.NewGuid();
            _stock = new Stock();
            _nbStock = nbStock;
        }
        public Order(string orderName,Stock stock, int nbStock, DateTime orderDate, Guid id ) : this(orderName, stock, nbStock)
        {
            _orderName = string.IsNullOrEmpty(orderName) ? throw new ArgumentNullException(nameof(orderName)) : orderName;

        }
        public Order(string orderName, Stock stock, int nbStock, DateTime orderDate) : this(orderName, stock, nbStock)
        {
            _id = Guid.NewGuid();
            _orderName = string.IsNullOrEmpty(orderName) ? throw new ArgumentNullException(nameof(orderName)) : orderName;

        }
        public void AddStocks(Stock stock)
        {
            if (!(stock is null))
            {
                _stock = stock;
              
            }
            else
            {
                throw new ArgumentException("The stock is not Defined");
            }

        }

        public void modifyStock(Stock stock,string newNameStock, string newNameEnterprise, int newnbStock)
        {
            if (_stock == stock)
            {
                _stock._name = newNameStock;
                _stock._entrepriseName = newNameEnterprise;
                this._nbStock = newnbStock;

            }
            else
            {
                throw new ArgumentException("The Stock Does not correspond to the Chosen One! Please Check!");
            }

        }
    }
    
}

File: ApplicationStock/projetAngular/website-angular/src/models/Order.ts
import { Guid } from "guid-typescript"
import { Stock } from "./Stock";
export class Order{

 
    private _id  : Guid = Guid.createEmpty();
    private _orderName : string = '';
    private _orderDate : Date = new Date();

  
    private _nbStock :number = 0
    private _stock : Stock =  new Stock(Guid.createEmpty(),'',5,'');

    constructor(_id : Guid,_orderName :string,_orderDate : Date,_stock : Stock,_nbStock : number){
        this._id = _id;
        this._orderDate= _orderDate;
        this.orderName = _orderName;
        this._stock = _stock;
        this._nbStock = _nbStock
    }

    

    /**
     * Getter nbStock
     * @return {number }
     */
	public get nbStock(): number  {
		return this._nbStock;
	}

    /**
     * Setter nbStock
     * @param {number } value
     */
	public set nbStock(value: number ) {
		this._nbStock = value;
	}
      /**
     * Getter stock
     * @return {Stock }
     */
	public get stock(): Stock  {
		return this._stock;
	}

    /**
     * Setter stock
     * @param {Stock } value
     */
	public set stock(value: Stock ) {
		this._stock = value;
	}
       /**
     * Getter orderName
     * @return {string }
     */
	public get orderName(): string  {
		return this._orderName;
	}

    /**
     * Setter orderName
     * @param {string } value
     */
	public set orderName(value: string ) {
		this._orderName = value;
	}
    /**
     * Getter orderDate
     * @return {Date }
     */
	public get orderDate(): Date  {
		return this._orderDate;
	}

    /**
     * Setter orderDate
     * @param {Date } value
     */
	public set orderDate(value: Date ) {
		this._orderDate = value;
	}
 
    /**
     * Getter $stock
     * @return {Stock }
     */
	public get $stock(): Stock  {
		return this._stock;
	}

    /**
     * Setter $stock
     * @param {Stock } value
     */
	public set $stock(value: Stock ) {
		this._stock = value;
	}

}